src/lib/authselect_util.c

```diff
@@ -80,19 +80,40 @@ trimline(const char *str, char **_trimmed)
 static errno_t
 read_textfile_internal(FILE *file, const char *filename, char **_content)
 {
-    ssize_t bytes_read;
+    size_t read_bytes;
     char *buffer;
-    size_t len;
+    long filelen;
     errno_t ret;
 
-    buffer = NULL;
-    bytes_read = getdelim(&buffer, &len, '\0', file);
-    if (bytes_read == -1) {
+    ret = fseek(file, 0, SEEK_END);
+    if (ret != 0) {
         ret = errno;
         goto done;
     }
 
+    filelen = ftell(file);
+    if (filelen == -1) {
+        ret = errno;
+        goto done;
+    }
+
+    rewind(file);
+
+    buffer = malloc_zero_array(char, filelen + 1);
```

Comment:
Don't you want to have some kind of upper limit to prevent from absurdly large files from starving all memory? I don't think any files authselect handles will be ever larger than a couple of kilobytes..